src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -669,6 +678,33 @@ internal sealed override OverriddenOrHiddenMembersResult OverriddenOrHiddenMembe
             }
         }
 
+        public override string MetadataName
+        {
+            get
+            {
+                return (object)_replacedBy == null ?
+                    base.MetadataName :
+                    GeneratedNames.ReplacedMemberName(Name, uniqueId: 0);
```

Comment:
I'm a bit confused about how the replaced and originals are linked together. Are they like a linked list, where each replace method links to and from the method it replaces, or does the original method have a list of all of the replacements, or both? Is it worth adding a comment summarizing the strategy somewhere?